### ISSUE TITLE
Database migrations run twice for nop.web.framework and nop.data assemblies

### ISSUE THREAD

**Issue by jafin** [nopSolutions/nopCommerce#7056]
nopCommerce version: `develop`

Steps to reproduce the problem: 

Fluent Database migrations appear to run twice for assemblies `nop.web.framework` and `nop.data`

In `ApplicationBuilderExtensions.StartEngineAsync` :

```c#
var migrationManager = engine.Resolve<IMigrationManager>();
var assembly = Assembly.GetAssembly(typeof(ApplicationBuilderExtensions));  // <- nop.web.framework
migrationManager.ApplyUpMigrations(assembly, MigrationProcessType.Update);
assembly = Assembly.GetAssembly(typeof(IMigrationManager));  // <-- nop.web.data
migrationManager.ApplyUpMigrations(assembly, MigrationProcessType.Update);
```

and also in `NopDbStartup.ConfigureServices`, the list of scanned assemblies (`mAssemblies`) includes both `nop.web.framework` and `nop.data`

```c#
   foreach (var assembly in mAssemblies)
                runner.ApplyUpSchemaMigrations(assembly);
```

Suggest removing the execution from  `StartEngineAsync` as it runs after NopDbStartup.

This would reduce startup time a small amount, since FluentMigration doesn't have to scan for scripts run.

**Comment by jafin** [comment]
@mahdisafe 
To clarify the migrations are scanned twice, but run once, because the migrations write to the version table `MigrationVersionInfo` the first time, the second time is a noop   This is for a pristine install, there are two places in the code I mentioned above, which scan the same assemblies twice, which just adds unnecessary time to app startup.

**Comment by skoshelev** [comment]
Hi, @mahdisafe. There is one small nuance that you are overlooking. The bottom line is that NopDbStartup launches only those migrations that must change the database schema, and these migrations must be completed before the DI is fully initialized (see the method [ApplyUpSchemaMigrations](https://github.com/nopSolutions/nopCommerce/blob/25e57f7b477f954adc8ce9a1b7a073612defabf7/src/Libraries/Nop.Data/Migrations/MigrationManager.cs#L111)). 
And then we run the rest of the migrations by the StartEngineAsync method, only when everything is ready for that.

Closed #7056
